tests/test_change_compute.py

```diff
@@ -0,0 +1,318 @@
+    return expected.issubset({p.metadata.name for p in pods.items})
+
+
+async def do_pod_ids_exist(core: CoreV1Api, namespace: str, pod_ids: Set[str]) -> bool:
+    pods = await core.list_namespaced_pod(namespace=namespace)
+    return bool(pod_ids.intersection({p.metadata.uid for p in pods.items}))
+
+
+@pytest.mark.k8s
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "set_initial_request_resources, cpu_request, memory_request, cpu_limit,"
+    "memory_limit",
+    [
+        # Test a change in cpu and memory with requests = limits
+        (False, 1, "5Gi", 1, "5Gi"),
+        # Test a change in cpu and memory from req = limits to req != limits
+        (False, 1, "3Gi", 3, "5Gi"),
+        # Test a change in cpu and memory from req != limits to req = limits
+        (True, 1, "5Gi", 1, "5Gi"),
+    ],
+)
+async def test_change_cpu_and_ram(
+    set_initial_request_resources,
+    cpu_request,
+    memory_request,
+    cpu_limit,
+    memory_limit,
+    faker,
+    namespace,
+    kopf_runner,
+    api_client,
+):
+    """
+    Tests that the cratedb resource changes in cpu/memory requests/limits are properly
+    passed on to the statefulset.
+    Note we cannot test the affinity in a test cluster.
+    """
+    body_changes = [
+        {
+            "op": "replace",
+            "path": "/spec/nodes/data/0/resources/limits/cpu",
+            "value": cpu_limit,
+        },
+        {
+            "op": "replace",
+            "path": "/spec/nodes/data/0/resources/limits/memory",
+            "value": memory_limit,
+        },
+    ]
+    if set_initial_request_resources or cpu_request != cpu_limit:
+        if set_initial_request_resources:
+            body_changes.append(
+                {
+                    "op": "replace",
+                    "path": "/spec/nodes/data/0/resources/requests/cpu",
+                    "value": cpu_request,
+                },
+            )
+        else:
+            body_changes.append(
+                {
+                    "op": "add",
+                    "path": "/spec/nodes/data/0/resources/requests",
+                    "value": {
+                        "cpu": cpu_request,
+                        "memory": memory_request or memory_limit,
+                    },
+                },
+            )
+
+    if set_initial_request_resources:
+        body_changes.append(
+            {
+                "op": "replace",
+                "path": "/spec/nodes/data/0/resources/requests/memory",
+                "value": memory_request or memory_limit,
+            },
+        )
+
+    coapi = CustomObjectsApi(api_client)
+    core = CoreV1Api(api_client)
+    name = faker.domain_word()
+    if set_initial_request_resources:
+        # Start a cluster with requests set to half the original limits
+        initial_requests = {"cpu": 1, "memory": "2Gi"}
+        host, password = await start_cluster(
+            name, namespace, core, coapi, 3, "5.0.0", resource_requests=initial_requests
```

Comment:
Would be better to only start a 1 node cluster here - it takes wayyyy longer for 3 nodes to mount PVCs/initialize/restart etc.